File: revision/remove_element.cpp
#include<iostream>
#include<vector>
using namespace std;
void input(vector<int>&nums){
   for(auto &i : nums){
       cin >> i;
   }
}
void display(vector<int>nums){
    for(auto const i : nums){
        cout << i << ' ' ;
    }
    cout << endl;
}
int removeElement(vector<int>&nums, int val){
    int i = 0;
    for(int j = 0; j < nums.size(); j++){
        if (nums[j] != val) {
            nums[i] = nums[j];
            i++;
        }
    }
    return i;
}
int main()
{
    int n, val;
    cin >> n;
    vector<int>nums(n);
    input(nums);
    cin >> val;
    cout << removeElement(nums, val) << endl;
    return 0;
}

File: revision/remove_element.py
from typing import List
def removeElement(nums: List[int], val: int)->int:
    i = 0
    for j in range(len(nums)):
        if nums[j] != val:
            nums[i] = nums[j]
            i+=1
    return i    
nums = list(map(int, input().split()))
val = int(input())
print(removeElement(nums, val))